mrmustard/physics/wires.py

```diff
@@ -169,242 +310,255 @@ def __init__(
         modes_in_ket: set[int] | None = None,
         classical_out: set[int] | None = None,
         classical_in: set[int] | None = None,
-    ) -> None:
-        self.args: tuple[set, ...] = (
-            modes_out_bra or set(),
-            modes_in_bra or set(),
-            modes_out_ket or set(),
-            modes_in_ket or set(),
-            classical_out or set(),
-            classical_in or set(),
+    ):
+        if any(
```

Comment:
I don't think we need this check since the type hints already make it explicit that's what we expect

Reply:
I think we need it because otherwise one can pass an unsorted tuple, or list. Like (3,2,1) and then the modes are screwed up. To guard against that we would have to cast to a set, but I'd rather not do that by default because then implicit stuff happens.

Reply:
But we type hint it as a set does that mean we should update the type hint?

Reply:
What change do you have in mind? Here the hint says `set[int] | None` and the method raises if it isn't that.

Reply:
Right the type hint says its `set | None` so we shouldn't be expecting something like an unsorted tuple or list to be passed in. If we want to support passing in an unsorted tuple or list then the type hint should be updated to reflect that. If not, then the check shouldn't be necessary as the type hint is already implying it is `set | None`.

Reply:
As I'm sure you know, the type hint does not enforce anything. Raising an exception in this case guards against running into bugs because of an ordered data structure is passed in when out of order. If this is your only concern about this PR can you accept this check and approve it?